src/orderly_web_api.py

```diff
@@ -0,0 +1,30 @@
+import requests
+
+
+class OrderlyWebAPI:
+    def __init__(self, base_url, token):
+        self.base_url = base_url
+        auth_url = self.build_url('login')
+        # data = 'grant_type=client_credentials'
+        headers = {'Authorization': 'token {}'.format(token)}
+        response = requests.post(auth_url, headers=headers)
+        if response.status_code != 200:
+            msg = 'Unexpected status code: {}. Unable to authenticate.'
+            raise Exception(msg.format(response.status_code))
+        self.token = response.json()['access_token']
+
+    def run_report(self, report, params):
```

Comment:
Do we need anything here to get the report status endpoint? If this is for the task queue, presumably it needs to know if the report is done or not?

Reply:
Yep, it's returning the key here, which can be used for polling. Or the client could handle polling and call a callback when the run is complete. But I thought I'd leave that for another ticket!

Reply:
It would be part of https://mrc-ide.myjetbrains.com/youtrack/issue/VIMC-4014